src/parallel_simulation/network/routing_kit_network.rs

```diff
@@ -0,0 +1,34 @@
+use rust_road_router::datastr::graph::{EdgeId, NodeId, Weight};
+
+#[derive(Debug, Clone)]
+pub struct RoutingKitNetwork {
+    //CSR graph representation
+    pub(crate) first_out: Vec<EdgeId>,
```

Comment:
all your members are `pub`, but you also have getters for `first_out`, `head` and `travel_time`. Why?

Reply:
Good Point 🙃